rd-net/RdFramework/Impl/SocketWire.cs

```diff
@@ -639,25 +643,32 @@ public Server(Lifetime lifetime, IScheduler scheduler, Socket serverSocket, stri
       private Server(Lifetime lifetime, IScheduler scheduler, string? optId = null) : base("ServerSocket-"+(optId ?? "<noname>"), lifetime, scheduler)
       {}
 
-      public static Socket CreateServerSocket(IPEndPoint? endPoint)
+      public static Socket CreateServerSocket(EndPointWrapper? endPointWrapper)
       {
-        Protocol.InitLogger.Verbose("Creating server socket on endpoint: {0}", endPoint);
+        Protocol.InitLogger.Verbose("Creating server socket on endpoint: {0}", endPointWrapper?.EndPointImpl);
+        // by default we will use IPEndpoint?
+        endPointWrapper ??= EndPointWrapper.CreateIpEndPoint(IPAddress.Loopback, 0);
 
-        var serverSocket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+        var serverSocket = new Socket(endPointWrapper.AddressFamily, endPointWrapper.SocketType, endPointWrapper.ProtocolType);
         SetSocketOptions(serverSocket);
 
-        endPoint = endPoint ?? new IPEndPoint(IPAddress.Loopback, 0);
-        serverSocket.Bind(endPoint);
+        if (endPointWrapper.LocalPath != null)
+        {
+          if (File.Exists(endPointWrapper.LocalPath)) File.Delete(endPointWrapper.LocalPath);
+        }
+        serverSocket.Bind(endPointWrapper.EndPointImpl);
         serverSocket.Listen(1);
-        Protocol.InitLogger.Verbose("Server socket created, listening started on endpoint: {0}", endPoint);
+        Protocol.InitLogger.Verbose("Server socket created, listening started on endpoint: {0}", endPointWrapper.EndPointImpl);
 
         return serverSocket;
       }
 
       private void StartServerSocket(Lifetime lifetime, Socket serverSocket)
       {
         if (serverSocket == null) throw new ArgumentNullException(nameof(serverSocket));
-        Port = ((IPEndPoint) serverSocket.LocalEndPoint).Port;
+        Port = serverSocket.LocalEndPoint is IPEndPoint ipEndPoint ? ipEndPoint.Port : null;
+        // TODO do we want to store Path from Unix here? Only via reflection.
```

Comment:
Perhaps let's keep the `Endpoint` or the `Socket` object somewhere in the public properties, and let the user to sort them out?